torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -41,6 +45,20 @@ std::shared_ptr<FutureMessage> py_rpc_builtin(
         Stack stack = torch::jit::createStackForSchema(
             op->schema(), args, kwargs, c10::nullopt);
 
+        auto& autogradContainer = DistAutogradContainer::getInstance();
+        if (autogradContainer.hasValidContext()) {
+          // Get all the ivalues for the operator.
+          auto ivalues =
+              torch::jit::last(stack, op->schema().arguments().size());
+
+          // Attach the appropriate autograd edges.
+          auto grad_fn = addSendRpcBackward(ivalues);
+
+          // Record the send function in our current context.
+          auto& currentContext = autogradContainer.currentContext();
+          currentContext.addSendFunction(grad_fn);
+        }
+
         return agent.send(
             dstName, ScriptCall(op, std::move(stack)).toMessage());
       } catch (std::runtime_error) {
```

Comment:
Also not part of this PR, but can we bound the try-catch block around the only scope which you actually expect to throw--the comment above implies `createStackForSchema` is the only thing that is currently throwing? This will help us from accidentally catching errors we don't intend to catch. It's very simple to do, while we're waiting for someone to actually refactor the downstream function in question. (Or you could just do that refactor?)